homeassistant/components/lyric/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for Honeywell Lyric."""
+import asyncio
+import logging
+
+import async_timeout
+from lyric import Lyric
+
+from homeassistant import config_entries
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.config_entries import ConfigFlow
+from .const import (
+    AUTH_CALLBACK_NAME,
+    AUTH_CALLBACK_PATH,
+    DOMAIN,
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_LYRIC_CONFIG_FILE,
+    DATA_LYRIC_CONFIG,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class LyricFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a Lyric config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize Lyric flow."""
+        self.lyric = None
+        self.client_id = None
+        self.client_secret = None
+        self.code = None
+
+    async def async_step_user(self, code=None):
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if not self.hass.data.get(DATA_LYRIC_CONFIG):
+            return self.async_abort(reason="no_config")
+
+        conf = self.hass.data.get(DATA_LYRIC_CONFIG)
+
+        self.client_id = conf[CONF_CLIENT_ID]
```

Comment:
Okay. You're probably want to shout at me when you read this, but I think I have a PR #27727 that will make it no longer needed to implement your own OAuth2 flow. Using this new helper, you will be able to get Home Assistant take care of account linking. You will just have to get the tokens passed to the Lyric lib afterwards once it is set up.

Reply:
Took a look at the somfy integration and replicated. Tested and it works perfectly:

![image](https://user-images.githubusercontent.com/28114703/67620205-4a302800-f7fc-11e9-9436-25e7ff694299.png)


It does save time and effort, just wish you'd had that helper there before I went to the initial effort :smile: